x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/AsyncWaitStep.java

```diff
@@ -28,11 +29,21 @@ protected Client getClient() {
         return client;
     }
 
-    public abstract void evaluateCondition(IndexMetaData indexMetaData, Listener listener);
+    public void evaluateCondition(Settings settings, IndexMetaData indexMetaData, Listener listener){
+        evaluateCondition(indexMetaData, listener);
+    }
+
+    public void evaluateCondition(IndexMetaData indexMetaData, Listener listener){
+        try {
+            throw new UnsupportedOperationException();
+        } catch (UnsupportedOperationException e) {
+            listener.onFailure(e);
+        }
```

Comment:
On second thought, would it make sense to have only one method? ie. add the `Settings` as parameter? Or even more, maybe pass the timeout directly (as currently that's the only reason we want to modify this signature) ?

Reply:
I think it would make more sense to change the single, abstract, method. We should be able to pass in the Settings where necessary and just ignore them on the steps that don't need them.

Reply:
I didn't want to change classes that don't need timeout and this parameter is useless for them. But you are right that this is messy, I refectored it back to single abstract method with `TimeValue masterTimeout` parameter as suggested by @andreidan